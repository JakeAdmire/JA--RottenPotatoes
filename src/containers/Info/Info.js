import React, { Component } from 'react';

export class Info extends Component {

  buildDate(date) {
    return date.split('T')[0];
  }

  buildDistance = (distance) => {
    let splitDistance = distance.toString().split('.');
    let decimal = splitDistance[1].slice(0, 1);

    return splitDistance[0] + '.' + decimal + 'm away';
  }

  render() {
    const { name, address, correctPlaces, distance } = this.props;
    const { inspectiontype, violation, violationtype, inspectionscore, inspectiondate } = correctPlaces[0];

    let date = this.buildDate(inspectiondate);
    let distanceText = this.buildDistance(distance);
    return(
      <div className="Info">
        <h1>{name}</h1>
        <p>{distanceText}</p>
        <p>{address}</p>
        <h3>Latest Inspection:</h3>
        <p>Date: {date}</p>
        <p>Inspection Type: {inspectiontype}</p>
        <p>Inspection Violation: {violation}</p>
        <p>Violation Status: {violationtype}</p>
        <p>Inspection Score: {inspectionscore}</p>
      </div>
    )
  }
}