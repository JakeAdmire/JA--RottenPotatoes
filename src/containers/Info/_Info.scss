.Info {
  box-sizing: border-box;
  background: #c58b6c;
  border-radius: 15px;
  padding: 10px;
}