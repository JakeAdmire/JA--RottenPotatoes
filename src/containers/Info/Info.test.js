import React from 'react';
import { Info } from './Info';
import { shallow } from 'enzyme';

describe('Info', () => {

  let wrapper;

  let mockProps = {
    name: 'TACO BELL', 
    address: '123 FAKE ST. FORT COLLINS, CO', 
    correctPlaces: [
      { inspectiondate: '1234543' }
    ], 
    distance: 12.3123
  };

  beforeEach(() => {
    wrapper = shallow(
      <Info {...mockProps} />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should return a manipulated string when buildDate is invoked', () => {
    const mockDate = '2019-12-04T000000';
    const buildDate = wrapper.instance().buildDate(mockDate);
    expect(buildDate).toEqual("2019-12-04");
  })

  it('should return a manipulated string when buildDistance is invoked', () => {
    const buildDistance = wrapper.instance().buildDistance(mockProps.distance);
    expect(buildDistance).toEqual("12.3m away");
  })

})