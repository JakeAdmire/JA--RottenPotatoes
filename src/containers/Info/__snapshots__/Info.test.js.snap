// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Info
    address="123 FAKE ST. FORT COLLINS, CO"
    correctPlaces={
      Array [
        Object {
          "inspectiondate": "1234543",
        },
      ]
    }
    distance={12.3123}
    name="TACO BELL"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          TACO BELL
        </h1>,
        <p>
          12.3m away
        </p>,
        <p>
          123 FAKE ST. FORT COLLINS, CO
        </p>,
        <h3>
          Latest Inspection:
        </h3>,
        <p>
          Date: 
          1234543
        </p>,
        <p>
          Inspection Type: 
        </p>,
        <p>
          Inspection Violation: 
        </p>,
        <p>
          Violation Status: 
        </p>,
        <p>
          Inspection Score: 
        </p>,
      ],
      "className": "Info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TACO BELL",
        },
        "ref": null,
        "rendered": "TACO BELL",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "12.3m away",
        },
        "ref": null,
        "rendered": "12.3m away",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "123 FAKE ST. FORT COLLINS, CO",
        },
        "ref": null,
        "rendered": "123 FAKE ST. FORT COLLINS, CO",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Latest Inspection:",
        },
        "ref": null,
        "rendered": "Latest Inspection:",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Date: ",
            "1234543",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Date: ",
          "1234543",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Inspection Type: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Inspection Type: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Inspection Violation: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Inspection Violation: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Violation Status: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Violation Status: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Inspection Score: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Inspection Score: ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            TACO BELL
          </h1>,
          <p>
            12.3m away
          </p>,
          <p>
            123 FAKE ST. FORT COLLINS, CO
          </p>,
          <h3>
            Latest Inspection:
          </h3>,
          <p>
            Date: 
            1234543
          </p>,
          <p>
            Inspection Type: 
          </p>,
          <p>
            Inspection Violation: 
          </p>,
          <p>
            Violation Status: 
          </p>,
          <p>
            Inspection Score: 
          </p>,
        ],
        "className": "Info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TACO BELL",
          },
          "ref": null,
          "rendered": "TACO BELL",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "12.3m away",
          },
          "ref": null,
          "rendered": "12.3m away",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "123 FAKE ST. FORT COLLINS, CO",
          },
          "ref": null,
          "rendered": "123 FAKE ST. FORT COLLINS, CO",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Latest Inspection:",
          },
          "ref": null,
          "rendered": "Latest Inspection:",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Date: ",
              "1234543",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Date: ",
            "1234543",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Inspection Type: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Inspection Type: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Inspection Violation: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Inspection Violation: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Violation Status: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Violation Status: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Inspection Score: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Inspection Score: ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
